File: back-flux/service/app/api/v1/schemas/session.py
from datetime import datetime

from pydantic import BaseModel, Field


class SessionResponse(BaseModel):
    is_session_need_update: bool = Field(
        ...,
        examples=[True],
    )

class SessionRefreshResponse(BaseModel):
    token: str = Field(
        examples=['24lsPl3AouvV8OkYtWpOpo']
    )


class SessionListItem(BaseModel):
    id: int
    device_id: str
    device_name: str | None = None
    platform: str | None = None
    app_version: str | None = None
    provider: str | None = None
    is_current: bool
    created_at: datetime
    expires_at: datetime


class SessionListResponse(BaseModel):
    sessions: list[SessionListItem]

File: front/lib/futures/profile/models/session.dart
class ProfileSession {
  final int id;
  final String deviceId;
  final String? deviceName;
  final String? platform;
  final String? appVersion;
  final String? provider;
  final bool isCurrent;
  final DateTime createdAt;
  final DateTime expiresAt;

  const ProfileSession({
    required this.id,
    required this.deviceId,
    required this.deviceName,
    required this.platform,
    required this.appVersion,
    required this.provider,
    required this.isCurrent,
    required this.createdAt,
    required this.expiresAt,
  });

  factory ProfileSession.fromJson(Map<String, dynamic> json) {
    return ProfileSession(
      id: json['id'] as int,
      deviceId: json['device_id']?.toString() ?? '',
      deviceName: json['device_name']?.toString(),
      platform: json['platform']?.toString(),
      appVersion: json['app_version']?.toString(),
      provider: json['provider']?.toString(),
      isCurrent: json['is_current'] == true,
      createdAt: DateTime.parse(json['created_at'].toString()).toLocal(),
      expiresAt: DateTime.parse(json['expires_at'].toString()).toLocal(),
    );
  }
}

class ProfileSessionsResponse {
  final List<ProfileSession> sessions;

  const ProfileSessionsResponse({
    required this.sessions,
  });

  factory ProfileSessionsResponse.fromJson(Map<String, dynamic> json) {
    final rawSessions = json['sessions'];

    if (rawSessions is! List) {
      return const ProfileSessionsResponse(sessions: []);
    }

    return ProfileSessionsResponse(
      sessions: rawSessions
          .whereType<Map>()
          .map((item) => ProfileSession.fromJson(
                Map<String, dynamic>.from(item),
              ))
          .toList(),
    );
  }
}
